src/drawing.lisp

```diff
@@ -109,6 +109,45 @@
   ;; TODO: Drawing in event handlers could be useful with COPY-PIXELS set to to T.
   (warn "Can't draw from current context (e.g. an event handler)."))
 
+(defun draw-polygon (points)
+  (let ((tobj (make-instance 'polygon-tessellator)))
+    (glu:tess-property tobj :winding-rule :odd)
+    (glu:with-tess-polygon (tobj)
+      (glu:with-tess-contour tobj
+        (loop for (x y) in points
+              ;; Expects 3d coordinates.
+              do (glu:tess-vertex tobj (list x y 0) (list x y)))))
+    (glu:tess-delete tobj)
+    (loop for (primitive . points) in (pt-shapes tobj)
+          do (draw-shape primitive points nil))
+    (draw-shape nil nil points)))
```

Comment:
Could we maybe leave a comment describing what these expressions are doing? They're a bit mysterious.

Reply:
Sure